logprep/processor/requester/rule.py

```diff
@@ -0,0 +1,186 @@
+        params: dict = field(
+            validator=[
+                validators.instance_of(dict),
+                validators.deep_mapping(
+                    key_validator=validators.instance_of(str),
+                    value_validator=validators.instance_of(str),
+                ),
+            ],
+            factory=dict,
+        )
+        """ (Optional) The query parameters as dictionary. Can be templated by using the pattern
+        :code:`${the.dotted.field}` somewhere in the key or value all elements."""
+        headers: dict = field(
+            validator=[
+                validators.instance_of(dict),
+                validators.deep_mapping(
+                    key_validator=validators.instance_of(str),
+                    value_validator=validators.instance_of(str),
+                ),
+            ],
+            factory=dict,
+        )
+        """ (Optional) The http headers as dictionary."""
+        auth: tuple = field(
+            validator=[validators.instance_of(tuple)],
+            converter=tuple,
+            factory=tuple,
+        )
+        """ (Optional) The authentication tuple. Defined as list. Will be converted to tuple"""
+        timeout: float = field(validator=validators.instance_of(float), converter=float, default=2)
+        """ (Optional) The timeout in seconds as float for the request. Defaults to 2 seconds"""
+        verify: bool = field(validator=validators.instance_of(bool), default=True)
+        """ (Optional) Wether or not verify the ssl context. Defaults to :code:`True`"""
+        proxies: dict = field(
+            validator=[
+                validators.instance_of(dict),
+                validators.deep_mapping(
+                    key_validator=validators.instance_of(str),
+                    value_validator=validators.instance_of(str),
+                ),
+            ],
+            factory=dict,
+        )
+        """(Optional) Dictionary mapping protocol or protocol and host to the
+        URL of the proxy (e.g. :code:`{"http": "foo.bar:3128", "http://host.name": "foo.bar:4012"}`)
+        to be used on the request"""
+        cert: str = field(validator=validators.instance_of(str), default="")
+        """(Optional) SSL client certificate as path to ssl client cert file (.pem)."""
+
+        def __attrs_post_init__(self):
+            url_fields = re.findall(FIELD_PATTERN, self.url)
+            json_fields = re.findall(FIELD_PATTERN, json.dumps(self.json))
+            data_fields = re.findall(FIELD_PATTERN, self.data)
+            params_fields = re.findall(FIELD_PATTERN, json.dumps(self.params))
+            self.source_fields = list({*url_fields, *json_fields, *data_fields, *params_fields})
+
+    # pylint: disable=missing-docstring
+    @property
+    def kwargs(self):
+        kwargs = {
+            key: getattr(self._config, key)
+            for key in REQUEST_CONFIG_KEYS
+            if getattr(self._config, key)
```

Comment:
Probably it won't make a difference, but couldn't you use here `hasattr`?

Reply:
no, because I only want to add if I get a truthy value back. I do not want to add empty fields